### PR TITLE
Feature/lol/group layer selection

### PR DESCRIPTION
**Pull request by lliss**
Connects #776 

 * Change layer settings from change for overlay layers in subgroups (raster and vector)
 * Extend leaflet's Layer control to use these special groups to create radios rather than checkboxes.
 * Enable a "no selection" option to disable all overlay layers.

To test:
 * Open the layer control.
 * Switch to the overlays tab.
 * Choose a vector layer.
 * Choose a raster layer.
 * Ensure you can disable both layers.
 * Ensure you cannot enable more than one of each type of layer.

### CODE REVIEW

File: src/mmw/js/src/core/views.js [hunk 1]
```diff
@@ -338,7 +351,7 @@ var MapView = Marionette.ItemView.extend({
                     leafletLayer = new L.Google(layer.googleType, {
                         maxZoom: layer.maxZoom
                     });
-                } else {
+                } else if (!layer.empty) {
```
Comment (rajadain) [inline src/mmw/js/src/core/views.js]: Apparently `layer` is undefined in some cases, causing all tests to fail.

**Comment by rajadain** [comment]
+1. Functionally works very well, and looks pretty nice:

image

I wish that there was more room above the "Coverage" heading, but bootstrap by default only gives those bottom paddings, not top. It will probably be addressed when Design takes a look at that entire box.

Code is great too, except for that one failing use case which is preventing tests.

Great work!

File: src/mmw/apps/home/views.py [hunk 2]
```diff
@@ -115,7 +115,8 @@ def get_client_settings(request):
             'base_layers': get_layer_config('basemap'),
             'stream_layers': get_layer_config('stream'),
             'boundary_layers': get_layer_config('boundary'),
-            'overlay_layers': get_layer_config('overlay'),
+            'vector_layers': get_layer_config('vector'),
+            'raster_layers': get_layer_config('raster'),
```
Comment (caseypt) [inline src/mmw/apps/home/views.py]: Consider adding the word "overlay" in the attribute names just so it's clear that's what they are.

File: src/mmw/js/src/core/views.js [hunk 3]
```diff
@@ -188,7 +188,20 @@ var MapView = Marionette.ItemView.extend({
...
+                empty: true
+            },
+            nullRasterLayer = {
+                display: 'nullRaster',
+                raster: true,
+                empty: true
+            },
+            vectorLayers = [nullVectorLayer].concat(settings.get('vector_layers')),
+            rasterLayers = [nullRasterLayer].concat(settings.get('raster_layers')),
+            overlayLayers = vectorLayers.concat(rasterLayers);
```
Comment (caseypt) [inline src/mmw/js/src/core/views.js]: Consider splitting this logic out into a separate function that's called in initialize and returns `overayLayers`, just for the sake of keeping the `initialize` function shorter (it was previously ~300 lines).

**Comment by caseypt** [comment]
+1. Works well. Nice job.

File: src/mmw/js/src/core/views.js (hunk 1 above)
Comment (lliss) [inline src/mmw/js/src/core/views.js]: Looking into that now.

**Pull request merged**